cmake/BuildLibUnwind.cmake

```diff
@@ -0,0 +1,98 @@
+
+macro(libunwind_build)
+    set(LIBUNWIND_SOURCE_DIR ${PROJECT_SOURCE_DIR}/third_party/libunwind)
+    set(LIBUNWIND_BUILD_DIR ${PROJECT_BINARY_DIR}/build/libunwind)
+    set(LIBUNWIND_BINARY_DIR ${LIBUNWIND_BUILD_DIR}/work)
+    set(LIBUNWIND_INSTALL_DIR ${LIBUNWIND_BUILD_DIR}/dest)
+
+    include(ExternalProject)
+    ExternalProject_Add(bundled-libunwind-project
+                        TMP_DIR ${LIBUNWIND_BUILD_DIR}/tmp
+                        STAMP_DIR ${LIBUNWIND_BUILD_DIR}/stamp
+                        SOURCE_DIR ${LIBUNWIND_SOURCE_DIR}
+                        BINARY_DIR ${LIBUNWIND_BINARY_DIR}
+                        INSTALL_DIR ${LIBUNWIND_INSTALL_DIR}
+
+                        DOWNLOAD_COMMAND ""
+
+                        CONFIGURE_COMMAND
+                        <SOURCE_DIR>/configure
+                        CC=${CMAKE_C_COMPILER}
+                        CXX=${CMAKE_CXX_COMPILER}
+                        --prefix=<INSTALL_DIR>
+                        # Bundled libraries are linked statically.
+                        --disable-shared
+                        # Ditto.
+                        --enable-static
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L122-L125.
+                        --disable-coredump
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L130-L133.
+                        --disable-ptrace
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L138-L141.
+                        --disable-setjmp
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L143-L145
+                        --disable-documentation
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L147-L149
+                        --disable-tests
+                        # By default libunwind provides a weak alias to
+                        # `backtrace` function: this can lead to a conflict with
+                        # glibc's `backtrace`, see https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L151-L153
+                        --disable-weak-backtrace
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L155-L157
+                        --disable-unwind-header
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L302-L317
+                        --disable-minidebuginfo
+                        # See https://github.com/libunwind/libunwind/blob/e07b43c02d5cf1ea060c018fdf2e2ad34b7c7d80/configure.ac#L319-L334
+                        --disable-zlibdebuginfo
```

Comment:
Just in case: we don't generate an executable with compressed debug info in our packages, right?